modules/javafx.graphics/src/main/java/com/sun/javafx/css/media/expression/RangeExpression.java

```diff
@@ -74,15 +101,21 @@ public final boolean equals(Object obj) {
 
         RangeExpression other = (RangeExpression)obj;
         return other.featureType == featureType
-            && other.featureValue.equals(featureValue);
+            && other.numberValue == numberValue
+            && Objects.equals(other.sizeValue, sizeValue);
     }
 
     @Override
     public final int hashCode() {
-        return Objects.hash(getClass(), featureType, featureValue);
+        return Objects.hash(getClass(), featureType, sizeValue, numberValue);
+    }
+
+    final String getFormattedValue() {
+        return sizeValue != null ? sizeValue.toString() : Double.toString(numberValue);
     }
 
     public interface Supplier {
-        RangeExpression rangeExpression(SizeQueryType featureType, Size featureValue);
+        RangeExpression getSizeExpression(SizeQueryType featureType, Size sizeValue);
```

Comment:
minor: could we use a name other than Supplier to avoid clashing with java.util.function one (it is confusing).

Reply:
ESupplier? RProvider?  but if the access is always qualified, we are good.